code/modules/antagonists/mind_flayer/powers/flayer_passives.dm

```diff
@@ -0,0 +1,363 @@
+// This file contains all of the mindflayer passives
+
+/datum/mindflayer_passive
+	var/name = "default dan"
+	///The text we want to show the player in the shop
+	var/purchase_text = "Oopsie daisies! No purchase text on this ability!"
+	///The text shown to the player character on upgrade
+	var/upgrade_text = "Uh oh someone forgot to add upgrade text!"
+	///The level of the passive, used for upgrading passives. Basic level is 1
+	var/level = 0
+	var/max_level = 1
+	///The mob who the passive affects, usually an IPC. Set in force_add_abillity
+	var/mob/living/owner
```

Comment:
Is there any case where the owner would not be a carbon/human? If not, you can probably typecast it here and save yourselves a bunch of checks later